src/SpacetimeDBClient.cs

```diff
@@ -205,6 +212,17 @@ enum CompressionAlgos : byte
         {
             None = 0,
             Brotli = 1,
+            Gzip = 2,
+        }
+
+        private static BinaryReader BrotliReader(Stream stream)
+        {
+            return new BinaryReader(new BrotliStream(stream, CompressionMode.Decompress));
```

Comment:
I think you could change the return signature to `StreamReader`, return the `BrotliStream` (or `GZipStream` below), and wrap the `BinaryReader` at the caller; then you can there use the switch to get the streamreader, outside the switch wrap it into binary reader and call bsatn read :)